emcc.py

```diff
@@ -768,6 +768,11 @@ def get_file_suffix(filename):
   def optimizing(opts):
     return '-O0' not in opts
 
+  # Request LLVM debug info if we need it for source maps, or if the user
+  # requested it. Cyberdwarf also requires it.
+  def need_llvm_debug_info(options):
+    return use_source_map(options) or options.requested_debug == '-g' or shared.Settings.CYBERDWARF
```

Comment:
What about `-g2` and `-g3`?  We don't need llvm debug info in those cases?   Maybe we can use a numeric comparison on options.debug_level since `-g` implies `-g3`?  How about `options.debug_level >= 3`?

Reply:
Sounds good, but it is a little more radical a change... I see some test suite errors, which I'll look into.

Reply:
OK, lgtm to this as it stands if you prefer.  We can followup later.